src/platform/nrfconnect/ConnectivityManagerImpl.cpp

```diff
@@ -35,15 +36,55 @@
 
 #if CHIP_DEVICE_CONFIG_ENABLE_THREAD
 #include <platform/internal/GenericConnectivityManagerImpl_Thread.ipp>
+#include <platform/OpenThread/OpenThreadUtils.h>
 #endif
 
-using namespace ::chip;
-using namespace ::chip::TLV;
+using namespace ::chip::Inet;
 using namespace ::chip::DeviceLayer::Internal;
 
 namespace chip {
 namespace DeviceLayer {
 
+namespace {
+CHIP_ERROR JoinLeaveMulticastGroup(net_if * iface, const Inet::IPAddress & address, bool join)
+{
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+    if (net_if_l2(iface) == &NET_L2_GET_NAME(OPENTHREAD))
+    {
+        const otIp6Address otAddress = ToOpenThreadIP6Address(address);
+        const auto handler           = join ? otIp6SubscribeMulticastAddress : otIp6UnsubscribeMulticastAddress;
+        otError error;
+
+        ThreadStackMgr().LockThreadStack();
+        error = handler(openthread_get_default_instance(), &otAddress);
+        ThreadStackMgr().UnlockThreadStack();
+
+        return MapOpenThreadError(error);
+    }
+#endif
+
```

Comment:
Is there any chance that we have L2 different than WIFI and we still end up in the code above? I mean, maybe we can just similarly:

`if (net_if_l2(iface) == &NET_L2_GET_NAME(WIFI))` *(not sure if this work out this way)

before registering multicasts.

Reply:
I had that in the previous version but then I thought that the same code could be used not only for Wi-Fi but also for Ethernet or other interfaces one could think of so the condition wouldn't add any value. But given that it's under `ifdef` anyway, I can re-add the condition  to make it less confusing.

Reply:
I see, so it's up to you then. Maybe a comment that the same code is valid for Ethernet is enough.

Reply:
ok, I added a comment.